dashboard/test/controllers/admin_users_controller_test.rb

```diff
@@ -130,4 +133,22 @@ class AdminUsersControllerTest < ActionController::TestCase
     assert_response :success
     assert_select '.container .alert-danger', 'User not found -- email not confirmed'
   end
+
+  test "undelete_user undeletes deleted user" do
+    sign_in @admin
+
+    post :undelete_user, {user_id: @deleted_student.id}
+
+    assert @deleted_student.deleted_at.nil?
+  end
+
+  test "undelete_user noops for (non-deleted) user" do
+    sign_in @admin
+    existing_updated_at = @student.updated_at
+
+    post :undelete_user, {user_id: @student.id}
+
+    new_updated_at = @student.updated_at
```

Comment:
I think you have to reload student here to actually test this (active record objects aren't automatically synced with the db when changes happen "somewhere else", and I think making requests in controller tests count as somewhere else)